src/vsphere_cpi/lib/cloud/vsphere/nsxt_provider.rb

```diff
@@ -118,42 +124,58 @@ def initialize(client, default_vif_type)
     def add_vm_to_nsgroups(vm, ns_groups)
       return if ns_groups.nil? || ns_groups.empty?
       return if nsxt_nics(vm).empty?
-
-      logger.info("Adding vm '#{vm.cid}' to NSGroups: #{ns_groups}")
-      nsgroups = retrieve_nsgroups(ns_groups)
-
       lports = logical_ports(vm)
-      nsgroups.each do |nsgroup|
-        logger.info("Adding LogicalPorts: #{lports.map(&:id)} to NSGroup '#{nsgroup.id}'")
-        grouping_obj_svc.add_or_remove_ns_group_expression(
-          nsgroup.id,
-          *to_simple_expressions(lports),
-          'ADD_MEMBERS'
-        )
+      logger.info("Adding vm '#{vm.cid}' to NSGroups: #{ns_groups}")
+      Bosh::Retryable.new(
+          tries: 5,
+          on: [NSXTSecGrpUpdateError]
```

Comment:
What's the default sleep behavior since you did not specify in the code?

Reply:
`retryable.rb` has sleep parameter set to default as an exponential backoff. So if we do not pass it explicitly in the `Bosh::Retryable.new()`, it will execute the default option function expontential sleeper.

```
def default_options
      {
        tries: 2,
        sleep: exponential_sleeper,
        on: [],
        matching: /.*/,
        ensure: Proc.new {},
      }
    end
```

```
def exponential_sleeper
  lambda { |tries, _| [2**(tries-1), 10].min } # 1, 2, 4, 8, 10, 10..10 seconds
end
```